mlflow/tracking/client.py

```diff
@@ -1527,6 +1527,103 @@ def log_table(
+        specified artifact_file in the specified run_ids. The extra_columns are columns that
+        are not in the table but are augmented with run information and added to the DataFrame.
+
+        :param experiment_id: The experiment ID to load the table from.
+        :param artifact_file: The run-relative artifact file path in posixpath format to which
+                            the table is saved (e.g. "dir/file.json").
+        :param run_ids: Optional list of run_ids to load the table from. If no run_ids are
+                        specified, the table is loaded from all runs in the current experiment.
+        :param extra_columns: Optional list of extra columns to add to the returned DataFrame.
+
+        :return: pandas.DataFrame
+
+        .. test-code-block:: python
+            :caption: Example with passing run_ids
+
+            import mlflow
+            import pandas as pd
+
+            client = MlflowClient()
+            loaded_table = client.load_table(
+                experiment_id="uuid-0",
+                artifact_file="qabot_eval_results.csv",
+                run_ids=[
+                    "68f0c4aed0b74fc7bff23d6c3be74fed",
+                    "da1f56b292254a81858410606618d61c",
+                ],
+                # Append a column containing the associated run ID for each row
+                extra_columns=["run_id"],
+            )
+
+        .. test-code-block:: python
+            :caption: Example with passing no run_ids
+
+            # Loads the table with the specified name for all runs in the given
+            # experiment and joins them together
+            loaded_table = mlflow.load_table(
+                experiment_id="uuid-0",
+                "qabot_eval_results.csv",
+                # Append the run ID and the parent run ID to the table
+                extra_columns=["run_id", "parent_run_id"],
+            )
+        """
+        import pandas as pd
+
+        subset_tag_value = json.dumps({"path": artifact_file, "type": "table"})
+
+        # Build the filter string
+        tag_query = f"tags.{MLFLOW_LOGGED_ARTIFACTS} LIKE '%{subset_tag_value}%'"
+        run_query = ""
+        if run_ids is not None:
+            list_run_ids = ",".join(run_ids)
+            run_query = f" and attributes.run_id IN ('{list_run_ids}')"
+
+        filter_string = tag_query + run_query
+        runs = mlflow.search_runs(experiment_ids=[experiment_id], filter_string=filter_string)
+        data = pd.DataFrame()
+
+        # TODO: Add parallelism support here
+        def get_artifact_data(run):
+            run_id = run.run_id
+            norm_path = posixpath.normpath(artifact_file)
+            artifact_dir = posixpath.dirname(norm_path)
+            artifact_dir = None if artifact_dir == "" else artifact_dir
+            existing_predictions = pd.DataFrame()
+
+            artifacts = [f.path for f in self.list_artifacts(run_id, path=artifact_dir)]
```

Comment:
```suggestion
            artifacts = [f.path for f in self.list_artifacts(run_id, path=artifact_dir) if f.is_dir]
```

We normally don't use `<name>.json` as a directory name though.